spec/models/bus_stop_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'spec_helper'
+
+describe BusStop do
+  describe 'decide_if_completed_by' do
+    let(:user) { create :user }
+    let(:stop) { create :bus_stop, :pending }
+    context 'bus stop completed attribute changed' do
+      context 'bus stop is completed' do
+        it 'assigns user to completed by' do
+          Timecop.freeze Date.today do
+            stop.update! accessible: true
+            stop.completed = true
+          end
+          stop.decide_if_completed_by user
+          expect(stop.completed_by).to be user
```

Comment:
I'm not sure about `be` here. The two `User` instances will definitely be the same in the database, but I'm not sure about strict object-level equality.